### ISSUE TITLE
chef-solo displays error "The system cannot find the path specified" on first provision

### ISSUE THREAD

**Issue by sneal** [WinRb/vagrant-windows#74]
INFO interface: info: Running chef-solo...
Running chef-solo...
DEBUG winrmcommunicator: powershell executing:
function which {
    $command = [Array](Get-Command $args[0] -errorAction SilentlyContinue)
    if($null -eq $command)
    {
      exit 1
    }
    write-host $command[0].Definition
    exit 0
}

function test ([Switch] $d, [String] $path) {
  if(Test-Path $path)
  {
    exit 0
  }
  exit 1
}

function chown {
  exit 0
}

function mkdir ([Switch] $p, [String] $path)
{
  if(Test-Path $path)
  {
    exit 0
  } else {
    New-Item $path -Type Directory -Force | Out-Null
  }
}

```
      $old = Get-ExecutionPolicy;
      Set-ExecutionPolicy Unrestricted -force;
      c:\tmp\vagrant-chef-1\cheftask.ps1;
      Set-ExecutionPolicy $old -force
```

 INFO interface: info: ERROR: 
ERROR: 
 INFO interface: info: The system cannot find the path specified.
The system cannot find the path specified.

**Comment by StevenArmstrong** [comment]
Any ideas what is causing this as it is proving a major annoyance when I am provisioning windows boxes and as a result I can't use proper error handling. Is there any short term work arounds?

**Comment by sneal** [comment]
I don't understand how this is a major annoyance, is it causing your provision to fail or something? Right now this doesn't cause any problems for me besides some red text in the console, so its pretty low priority.

**Comment by StevenArmstrong** [comment]
I am scheduling the provisioning as a build step in Jenkins as part of a deployment pipeline, I then need to execute subsequent jobs after the provisioning step. However, as the provisioning step is throwing an exit code of 1 due to this issue, Jenkins throws an error on this step as it has an exit code >0. At the moment I am having to switch on continue on error for this step, but now my deployment pipeline will not fail if I have a genuine error on this step, which is not a good solution. I hope that makes sense.

**Comment by sneal** [comment]
Thanks for explaining. I don't think _this_ is your issue, I have a bunch of TeamCity builds setup that run the Chef provisioner just fine.

Are you using vagrant-windows 1.2.3? Failing an upgrade, can you send me the debug log output of one of your failed runs?

**Comment by StevenArmstrong** [comment]
Hi Shawn,

Yes I am using vagrant-windows 1.2.3.
I have attached my build log. I am executing vagrant from a standard exec
command in msbuild. Jenkins is just running the msbuild script which is
running the vagrant up command line.
Thanks for your help on this :)

Kind Regards,
                     Steven

**Comment by sneal** [comment]
I don't see a log attached

**Comment by sneal** [comment]
Fixed in 1.3.0
